packages/notification-center-vue/src/lib/notification-center-component/NotificationCenterComponent.vue

```diff
@@ -1,53 +1,143 @@
+    bellButtonClass,
+    gradientDotClass,
+    bellColors,
+  };
+});
+
+const updateArrowStyles = (arrowClass: string) => {
+  // add the popover arrow class to the arrow element, there is no better way to do this
+  const arrowClasses = ['v-popper__arrow-outer', 'v-popper__arrow-inner'];
+  const arrowsContainer = popper.value.$refs.popper.$_arrowNode as HTMLDivElement;
+  arrowsContainer.childNodes.forEach((node, idx) => {
+    (node as HTMLDivElement).className = '';
+    (node as HTMLDivElement).classList.add(arrowClasses[idx], arrowClass);
+  });
+};
+
+onMounted(() => {
+  updateArrowStyles(computedStyles.value.popoverArrowClass);
+
+  // listen to the unseen count changed event propagated from the web component
+  document.addEventListener('novu:unseen_count_changed', (event) => {
+    unseenCount.value = (event as CustomEvent).detail as number;
+  });
+});
+
+watch(computedStyles, (newComputedStyles) => {
+  updateArrowStyles(newComputedStyles.popoverArrowClass);
+});
 </script>
 
 <template>
-  <notification-center-component
-    :backendUrl="backendUrl"
-    :socketUrl="socketUrl"
-    :subscriberId="subscriberId"
-    :applicationIdentifier="applicationIdentifier"
-    :subscriberHash="subscriberHash"
-    :stores="stores"
-    :tabs="tabs"
-    :showUserPreferences="showUserPreferences"
-    :popover="popover"
-    :theme="theme"
-    :styles="styles"
-    :colorScheme="colorScheme"
-    :i18n="i18n"
-    :sessionLoaded="sessionLoaded"
-    :notificationClicked="notificationClicked"
-    :unseenCountChanged="unseenCountChanged"
-    :actionClicked="actionClicked"
-    :tabClicked="tabClicked"
-  />
+  <VDropdown
+    :theme="colorScheme"
+    :popperClass="computedStyles.popoverDropdownClass"
+    :placement="popover?.position"
+    :distance="popover?.offset"
+    ref="popper"
+  >
+    <!-- Popover target - usually button -->
+    <slot v-if="hasSlot" v-bind="unseenCount" :unseen-count="unseenCount" />
+    <BellButton
+      v-else
+      :bellButtonClass="computedStyles.bellButtonClass"
+      :gradientDotClass="computedStyles.gradientDotClass"
+      :unseenCount="unseenCount"
+      :colors="computedStyles.bellColors"
+    />
+    <!-- Popover content -->
+    <template #popper>
+      <notification-center-content-component
+        :backendUrl="backendUrl"
+        :socketUrl="socketUrl"
+        :subscriberId="subscriberId"
+        :applicationIdentifier="applicationIdentifier"
+        :subscriberHash="subscriberHash"
+        :stores="stores"
+        :tabs="tabs"
+        :showUserPreferences="showUserPreferences"
+        :theme="theme"
+        :styles="styles"
+        :colorScheme="colorScheme"
+        :i18n="i18n"
+        :sessionLoaded="sessionLoaded"
+        :notificationClicked="notificationClicked"
+        :unseenCountChanged="unseenCountChanged"
+        :actionClicked="actionClicked"
+        :tabClicked="tabClicked"
+      />
+    </template>
+  </VDropdown>
 </template>
+
+<style>
```

Comment:
no need in scoped styles because these class names are unique... with scoped styles the data attributes are not properly applied on the popover elements...